### PR TITLE
Better organization of collisions events

### PR DESCRIPTION
**Pull request by costinvisan**
### Contains
A better organization for collisions events and add some documentation

### How to test
Should work as before. Get in game and collide to check the collision properties.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by pollend** [comment]
I'm guessing this is just a rework of the names and none of the logic is different. if that is the case then this looks pretty good just from a glance.

File: engine/src/main/java/org/terasology/logic/characters/CharacterSoundSystem.java [hunk 1]
```diff
@@ -133,6 +144,33 @@ public void onLanded(VerticalCollisionEvent event, EntityRef entity, CharacterSo
         }
     }
 
+    /**
+     * React to any horziontal collison of a character that has a
+     * component for character sounds and decide what the volume
+     * of the landing sound should be and the sounds should pe played.
```
Comment (eviltak) [inline engine/src/main/java/org/terasology/logic/characters/CharacterSoundSystem.java]: Typo: `pe played` -> `be played`

File: modules/Core/src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java [hunk 2]
```diff
@@ -227,9 +227,15 @@ public void onHeal(DoHealEvent event, EntityRef entity) {
         checkHeal(entity, event.getAmount(), event.getInstigator());
     }
 
-    // TODO: This should be in a separate system
+    /**
+     * React to any vertical collision of an entity that has a
+     * health component and establishes if the character should
+     * take damage and what's the amount
```
Comment (eviltak) [inline modules/Core/src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java]: `     * take damage and in what amount.`

File: modules/Core/src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java [hunk 3]
```diff
@@ -227,9 +227,15 @@ public void onHeal(DoHealEvent event, EntityRef entity) {
...
+     * @param event the event that triggered the vertical collision
+     * @param entity the entity that collide
```
Comment (eviltak) [inline modules/Core/src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java]: Proper sentences please!

File: engine/src/main/java/org/terasology/logic/characters/CharacterSoundSystem.java [hunk 4]
```diff
@@ -133,6 +144,33 @@ public void onLanded(VerticalCollisionEvent event, EntityRef entity, CharacterSo
...
+     * Then send a PlaysSoundEvent
+     * @param event the event that triggered the horizontal collision
+     * @param entity the entity that collide horizontal
+     * @param characterSounds the component that contains info about character sounds
```
Comment (eviltak) [inline engine/src/main/java/org/terasology/logic/characters/CharacterSoundSystem.java]: Full sentences please!

File: engine/src/main/java/org/terasology/logic/characters/CharacterSoundSystem.java [hunk 5]
```diff
@@ -105,8 +107,17 @@ public void onJump(JumpEvent event, EntityRef entity, CharacterSoundComponent ch
...
     }
 
+    /**
+     * React to any vertical collison of a character that has
+     * a component for character sounds and decide what the volume
+     * of the landing sound should be and the sounds should pe played.
+     * Then send a PlaysSoundEvent
+     * @param event the event that triggered the vertical collision
+     * @param entity the entity that collide vertical
+     * @param characterSounds the component that contains info about character sounds
```
Comment (eviltak) [inline engine/src/main/java/org/terasology/logic/characters/CharacterSoundSystem.java]: `@param characterSounds The component that contains information on the character sounds.` 

Can we use full sentences with punctuation and the first letter capital not just here, but everywhere (including the two cases right above this one)? It makes things much more natural to read, which is very important in documentation :)

File: modules/Core/src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java [hunk 6]
```diff
@@ -241,8 +247,15 @@ public void onLand(VerticalCollisionEvent event, EntityRef entity) {
         }
     }
 
+    /**
+     * React to any horrizontal collision of an entity that has a
+     * health component and establishes if the character should
+     * take damage and what's the amount
```
Comment (eviltak) [inline modules/Core/src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java]: `     * take damage and in what amount.`

File: modules/Core/src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java [hunk 7]
```diff
@@ -241,8 +247,15 @@ public void onLand(VerticalCollisionEvent event, EntityRef entity) {
...
+     * @param event the event that triggered the vertical collision
+     * @param entity the entity that collide
```
Comment (eviltak) [inline modules/Core/src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java]: Full sentences :)

File: engine/src/main/java/org/terasology/logic/characters/CharacterSoundSystem.java [hunk 8]
```diff
@@ -133,6 +144,33 @@ public void onLanded(VerticalCollisionEvent event, EntityRef entity, CharacterSo
...
+     * @param entity the entity that collide horizontal
+     * @param characterSounds the component that contains info about character sounds
+     */
+    @ReceiveEvent
+    public void onHorrizontalCollision(HorizontalCollisionEvent event, EntityRef entity, CharacterSoundComponent characterSounds) {
+        Vector3f horizVelocity = new Vector3f(event.getVelocity());
+        horizVelocity.y = 0;
+        float velocity = horizVelocity.length();
+
+        if (velocity > HORIZONTAL_DAMAGE_SPEED_THRESHOLD) {
```
Comment (eviltak) [inline engine/src/main/java/org/terasology/logic/characters/CharacterSoundSystem.java]: I noticed that this was initially a reference to the `healthComponent.horizontalDamageSpeedThreshold` field, but has now been made a class-level constant -- can you elaborate on the reasons for this change?

**Comment by eviltak** [comment]
@costinvisan will you still be working on this? We'd love to get this merged as soon as possible, we really appreciate your work!

File: modules/Core/src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java [hunk 9]
```diff
@@ -257,24 +270,6 @@ public void onCrash(HorizontalCollisionEvent event, EntityRef entity) {
...
-            if (characterSounds.lastSoundTime + CharacterSoundSystem.MIN_TIME < time.getGameTimeInMs()) {
-                StaticSound sound = random.nextItem(characterSounds.landingSounds);
-                if (sound != null) {
-                    entity.send(new PlaySoundEvent(sound, characterSounds.landingVolume));
-                    characterSounds.lastSoundTime = time.getGameTimeInMs();
-                    entity.saveComponent(characterSounds);
-                }
-            }
-        }
-    }
```
Comment (eviltak) [inline modules/Core/src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java]: Any reason as to why this event handler has been removed? Looks like it is doing something different than the other one.

**Comment by 4Denthusiast** [comment]
This PR is mostly out-of-date by now.

**Pull request closed without merging**